tests/test_precision.py

```diff
@@ -147,37 +149,7 @@ def test_snippets():
         cmp_regimes,
     ) = stumpy.snippets(T, m, k, s=s, mpdist_T_subseq_isconstant=isconstant_custom_func)
 
-    # Revise fastmath flag, recompile, and re-calculate snippets,
-    # and then revert the changes
-    config.STUMPY_FASTMATH_FLAGS = {"nsz", "arcp", "contract", "afn"}
-    core._calculate_squared_distance.targetoptions["fastmath"] = (
-        config.STUMPY_FASTMATH_FLAGS
-    )
-    njit_funcs = cache.get_njit_funcs()
-    for module_name, func_name in njit_funcs:
-        module = importlib.import_module(f".{module_name}", package="stumpy")
-        func = getattr(module, func_name)
-        func.recompile()
-
-    (
-        cmp_snippets_NOreassoc,
-        cmp_indices_NOreassoc,
-        cmp_profiles_NOreassoc,
-        cmp_fractions_NOreassoc,
-        cmp_areas_NOreassoc,
-        cmp_regimes_NOreassoc,
-    ) = stumpy.snippets(T, m, k, s=s, mpdist_T_subseq_isconstant=isconstant_custom_func)
-
-    config._reset("STUMPY_FASTMATH_FLAGS")
-    core._calculate_squared_distance.targetoptions["fastmath"] = (
-        config.STUMPY_FASTMATH_FLAGS
-    )
-    for module_name, func_name in njit_funcs:
-        module = importlib.import_module(f".{module_name}", package="stumpy")
-        func = getattr(module, func_name)
-        func.recompile()
-
-    if np.allclose(ref_snippets, cmp_snippets):
+    if np.allclose(ref_snippets, cmp_snippets) or numba.config.DISABLE_JIT:
```

Comment:
Revising fastmath flag does not work when JIT is disabled. So, I revised this if condition to address that issue. However, the condition looks strange as it mixes two different things. IMO, there are two alternative options:

(1) Create two separate test functions, one for unit testing and one for coverage where JIT is disabled. So, we can do something like this:

```
# in tests/test_precision.py

def test_snippets_UnitTest:
     if numba.config.DISABLE_JIT: 
           pytest.skip("Skipping Tests No GPUs Available")
     # test precision

def test_snippets_Coverage:
     if not numba.config.DISABLE_JIT: 
           pytest.skip("Skipping Tests No GPUs Available")
     # test coverage
```

(2) Or, just exclude `tests/test_precision.py` when checking for coverage. Because the purpose of `tests/test_precison.py` is not about the coverage. So, we may revise the coverage part of the `test.sh` and do something like this:

```
for testfile in tests/test_*.py;
      if [[ $testfile -ef "tests/test_precision.py" ]]; then
        continue
      fi
      do
          # do coverage
      done

```

Reply:
Hmm, so I think you have a lot of repeated code and that's making things look weird. I am wondering if you could simplify it with:

```
    if ! np.allclose(ref_snippets, cmp_snippets):
        # Maybe you need `if not np.allclose(ref_snippets, cmp_snippets) and not numba.config.DISABLE_JIT:`??
        # Revise fastmath flag, recompile, and re-calculate snippets,
        # and then revert the changes

        # Also note that we've simply removed `reassoc` from our `fastmath` flags

        config.STUMPY_FASTMATH_FLAGS = {"nsz", "arcp", "contract", "afn"}
        core._calculate_squared_distance.targetoptions["fastmath"] = (
            config.STUMPY_FASTMATH_FLAGS
        )

        njit_funcs = cache.get_njit_funcs()
        for module_name, func_name in njit_funcs:
            module = importlib.import_module(f".{module_name}", package="stumpy")
            func = getattr(module, func_name)
            func.recompile()

        (
            cmp_snippets,
            cmp_indices,
            cmp_profiles,
            cmp_fractions,
            cmp_areas,
            cmp_regimes,
        ) = stumpy.snippets(
            T, m, k, s=s, mpdist_T_subseq_isconstant=isconstant_custom_func
        )
```

Rather than "NOreassoc", notice that I've left the return variables as 

```
        cmp_snippets,
        cmp_indices,
        cmp_profiles,
        cmp_fractions,
        cmp_areas,
        cmp_regimes,
```

And then after this, you can "blindly" perform your regular assertion checks:

```
    npt.assert_almost_equal(
            ref_snippets, cmp_snippets, decimal=config.STUMPY_TEST_PRECISION
        )
        npt.assert_almost_equal(
            ref_indices, cmp_indices, decimal=config.STUMPY_TEST_PRECISION
        )
        npt.assert_almost_equal(
            ref_profiles, cmp_profiles, decimal=config.STUMPY_TEST_PRECISION
        )
        npt.assert_almost_equal(
            ref_fractions, cmp_fractions, decimal=config.STUMPY_TEST_PRECISION
        )
        npt.assert_almost_equal(
            ref_areas, cmp_areas, decimal=config.STUMPY_TEST_PRECISION
        )
        npt.assert_almost_equal(ref_regimes, cmp_regimes)
```

Does that help clean up the logic?

Reply:
If anything I'm trying to remove as much of the complicated branching logic as possible

Reply:
This is definitely cleaner. The only downside is that if any of those assertion except the first one fails in unit testing, it is not possible for us to know whether it was from the case with `reassoc` or from the case without `reassoc`. 

And, based on what you wrote

> without needing to think about whether this is "NOreassoc" or not

I believe you are okay with it. This is a good trade-off.

Reply:
@seanlaw 

> \# Maybe you need `if not np.allclose(ref_snippets, cmp_snippets) and not numba.config.DISABLE_JIT:`??

Although `if not np.allclose(ref_snippets, cmp_snippets)` should suffice here (because if JIT is disabled, the value of `np.allclose(ref_snippets, cmp_snippets)` is True and the if will be skipped), it is safer to add ` and not numba.config.DISABLE_JIT` to the condition.